samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemRemoveListener.java

```diff
@@ -0,0 +1,145 @@
+import org.sakaiproject.tool.assessment.ui.listener.util.ContextUtil;
+
+
+/**
+ * <p>Title: Samigo</p>
+ * <p>Description: Sakai Assessment Manager</p>
+ * <p>Organization: Sakai Project</p>
+ */
+
+public class ItemRemoveListener implements ActionListener
+{
+  private static Logger log = LoggerFactory.getLogger(ItemModifyListener.class);
+  //private String scalename;  // used for multiple choice Survey
+
+  /**
+   * Standard process action method.
+   * @param ae ActionEvent
+   * @throws AbortProcessingException
+   */
+  public void processAction(ActionEvent ae) throws AbortProcessingException
+  {		
+	  ItemAuthorBean item = (ItemAuthorBean) ContextUtil.lookupBean("itemauthor");
+	  ItemService delegate = new ItemService();
+	  if(item.getItemToDelete() == null) return;
+	  Long deleteId= item.getItemToDelete().getItemId();
+	  ItemFacade itemf = delegate.getItem(String.valueOf(deleteId));
+	  // save the currSection before itemf.setSection(null), used to reorder question sequences
+	  SectionFacade  currSection = (SectionFacade) itemf.getSection();
+	  Integer  currSeq = itemf.getSequence();
+
+	  QuestionPoolService qpdelegate = new QuestionPoolService();
+	  if (qpdelegate.getPoolIdsByItem(deleteId.toString()) ==  null || qpdelegate.getPoolIdsByItem(deleteId.toString()).isEmpty()){
+		  // if no reference to this item at all, ie, this item is created in assessment but not assigned to any pool
+
+		  AuthorizationBean authzBean = (AuthorizationBean) ContextUtil.lookupBean("authorization");
+		  AssessmentService assessdelegate = new AssessmentService();
+		  AssessmentFacade af = assessdelegate.getBasicInfoOfAnAssessmentFromSectionId(currSection.getSectionId());
+		  if (!authzBean.isUserAllowedToEditAssessment(af.getAssessmentBaseId().toString(), af.getCreatedBy(), false)) {
+		      throw new IllegalArgumentException("User does not have permission to delete item in assessment: " + af.getAssessmentBaseId());
+		  }
+
+		  delegate.deleteItem(deleteId, AgentFacade.getAgentString());
+	  }
+	  else {
+		  if (currSection == null) {
+			  // if this item is created from question pool
+			  QuestionPoolBean  qpoolbean= (QuestionPoolBean) ContextUtil.lookupBean("questionpool");
+	          ItemFacade itemfacade= delegate.getItem(String.valueOf(deleteId));
+	          ArrayList items = new ArrayList();
+	          items.add(itemfacade);
+	          qpoolbean.setItemsToDelete(items);
+			  qpoolbean.removeQuestionsFromPool();
+			  return;
+		  }
+		  else {
+			  // 
+			  // if some pools still reference to this item, ie, this item is 
+			  // created in assessment but also assigned a a pool
+			  // then just set section = null
+			  itemf.setSection(null);
+			  delegate.saveItem(itemf);
+		  }
+	  }
+	//An item has been deleted
+	EventTrackingService.post(EventTrackingService.newEvent("sam.assessment.item.delete", "/sam/" +AgentFacade.getCurrentSiteId() + "/removed itemId=" + deleteId, true));
```

Comment:
Hey Brian.

Thanks for the feedback. I reused the original code from the bean.

Ill clean it up and let you know when it's done.

Fg
On Jun 3, 2016 8:50 AM, "Brian Jones" <notifications@github.com> wrote:

> In
> samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemRemoveListener.java
> <https://github.com/sakaiproject/sakai/pull/2735#discussion_r65701480>:
>
> > +	          items.add(itemfacade);
> > +	          qpoolbean.setItemsToDelete(items);
> > +			  qpoolbean.removeQuestionsFromPool();
> > +			  return;
> > +		  }
> > +		  else {
> > +			  //
> > +			  // if some pools still reference to this item, ie, this item is
> > +			  // created in assessment but also assigned a a pool
> > +			  // then just set section = null
> > +			  itemf.setSection(null);
> > +			  delegate.saveItem(itemf);
> > +		  }
> > +	  }
> > +	//An item has been deleted
> > +	EventTrackingService.post(EventTrackingService.newEvent("sam.assessment.item.delete", "/sam/" +AgentFacade.getCurrentSiteId() + "/removed itemId=" + deleteId, true));
>
> Same here, tabs vs. spaces
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sakaiproject/sakai/pull/2735/files/4b843c527f41ce1c826bf8f053ac17f9e1278ce9#r65701480>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEnF-HA3Y6HxtU0jlfbEUpNMp6o8TJUxks5qICMggaJpZM4Is8hZ>
> .
>